### PR TITLE
#184637135 feat:google-login-authentication

### PR DESCRIPTION
**Pull request by buka1calvin**
#### What does this PR do?
As a user, I can login in to the app  via Google.
#### Description of Task to be completed?
Currently as a new user cannot login with their pre-existing social media accounts. He/She can only login with email password combinations.
as a new user i should be able to login with google 
* here are some end points you can use for testing
  `GET /api/v1/users`:for getting the path to the google login form
  `GET /api/v1/users/login`:here you can login directly
  `GET /api/v1/users/google/callback`:this is the link that will direct you after a successful login
   `GET /api/v1/users/protected`:this is the page you're directed in
   `GET /api/v1/users/failure`:this is the page redirected when the the actual page is not available
   
#### How should this be manually tested?
after cloning the repo, cd into it and RUN : `npm run dev`
then you can  test the endpoints provided above using postman or your browser preferably

#### Any background context you want to provide?
this is the google authentication for users 

#### What are the relevant pivotal tracker/Trello stories
#184637138

### CODE REVIEW

**Review by mjeanbosco19 — CHANGES_REQUESTED** [review]
@buka1calvin 
Resolve conflict

**Review by katros1 — APPROVED** [review]

File: src/services/googleAuth.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  async (request, accessToken, refreshToken, profile, done)=> {
+    try {
+        // const [user] = await User.findOrCreate({
+        //   where: { googleId: profile.id },
+        //   defaults: {                
+        //     name: profile.displayName,
+        //     email: profile.emails[0].value,
+        //   },
+        // });
+        // done(null, user);
```
Comment (Abdulkeza) [inline src/services/googleAuth.js]: remove this

**Review by nkurunziza1 — APPROVED** [review]

**Review by nkurunziza1 — APPROVED** [review]

File: src/services/googleAuth.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        const existingUser = await User.User.findOne({ where: {email: profile.emails[0].value} });                            
+                            if (existingUser) {
+                                // res.send(null, existingUser);
+                                const {firstname,email,role}=existingUser
+                                const person={firstname,email,role}
+              const token=  generateToken(person)
+               console.log(token)
```
Comment (Abdulkeza) [inline src/services/googleAuth.js]: remove console log

File: src/services/googleAuth.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                                return done(null, existingUser);
+                            }
+                            // else {
+                            //   // res.send(`<h3>this is the end</h3>`)
+                            //   console.log('user not found')
+                            // }
```
Comment (Abdulkeza) [inline src/services/googleAuth.js]: remove this comments

File: src/services/googleAuth.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+));
+passport.serializeUser((user, done) => {
+    done(null, user.email);
+  });
+  
+  // passport.deserializeUser((id, done) => {
+  //   User.findById(_id).then(user => {
+  //     done(null, user);
+  //   });
+  //   });
```
Comment (Abdulkeza) [inline src/services/googleAuth.js]: remove this

**Review by leandreAlly — APPROVED** [review]

**Review by Abdulkeza — APPROVED** [review]

**Pull request merged**